github/helpers/githubSDK.ts

```diff
@@ -726,3 +726,20 @@ export async function updateGithubIssues(
     }
     return repsonseJSON;
 }
+
+export async function isRepositoryExist(
+    http: IHttp,
+    repoName: string
+): Promise<boolean> {
+    const response = await http.get(`${BaseRepoApiHost}${repoName}`, {
+        headers: {
+            "Content-Type": "application/json",
+        },
+    });
```

Comment:
@Nabhag8848 Handling the endpoint for unauthorized users should happen conditionally. This will return false for any private repo. We should ideally try get the access token, if it exists we add it to the header else we just skip it.

Reply:
[This code snippet should help](https://github.com/RocketChat/Apps.Github22/blob/main/github/helpers/githubSDK.ts#L664-L677).

Reply:
Yep @samad-yar-khan , will fix this rn.
 i found another issue within this case when running command having private repo **without login**. So our message says `Hey {sender} Provided username and/or repository doesn't exist` even though the repo exist actually but user is not loggedIn. This leads to we will need to change the message to `Hey {sender} Provided repository doesn't exist or you are not logged in`

Reply:
@Nabhag8848 We can show `Hey {sender} Provided repository doesn't exist or you need to login to access private repo.` incase the user is not logged in but if the token is there we can always show `Hey {sender} Provided username and/or repository doesn't exist`